scripts/hourly_wmb.py

```diff
@@ -6,174 +6,16 @@
-            for row in reader:
-                data.append(row)
-            log.info('instantiated the reader and processed all rows')
-        except ftplib.all_errors as e:
-            log.critical('Unable to load data from ftp source', exc_info=True)
-            sys.exit(1)
-
-        # save the downloaded file
-        fname_out = os.path.join(args.cache_dir, 'wmb_download' + datetime.strftime(datetime.now(), '%Y-%m-%dT%H-%M-%S') + '.csv')
-        with open(fname_out, 'w') as f_out:
-            copier = csv.DictWriter(f_out, fieldnames = reader.fieldnames)
-            copier.writeheader()
-            copier.writerows(data)
-
-    log.info('{0} observations read into memory'.format(len(data)))
-    dl = DataLogger()
-    
-    # Open database connection
-    try:
-        engine = create_engine(args.connection_string)
-        Session = sessionmaker(engine)
-        sesh = Session()
-        sesh.begin_nested()
-    except Exception as e:
-        dl.add_row(data, 'db-connection error')
-        log.critical('''Error with Database connection 
-                            See logfile at {l}
-                            Data saved at {d}
-                            '''.format(l = args.log, d = data_archive), exc_info=True)
-        sys.exit(1)
-
-    try:
-        op = ObsProcessor(sesh, data, args)
-        op.process()
-        if args.diag:
-            log.info('Diagnostic mode, rolling back all transactions')
-            sesh.rollback()
-        else:
-            log.info('Commiting the session')
-            sesh.commit()
-            
-    except Exception as e:
-        dl.add_row(data, 'preproc error')
-        sesh.rollback()
-        data_archive = dl.archive(args.archive_dir)
-        log.critical('''Error data preprocessing. 
-                            See logfile at {l}
-                            Data saved at {d}
-                            '''.format(l = args.log, d = data_archive), exc_info=True)
-        sys.exit(1)
-    finally:
-        sesh.commit()
-        sesh.close()
-
-class FTPReader(object):
-    '''Glue between the FTP_TLS class methods (which are callback based)
-       and the csv.DictReader class (which is iteration based)
-    '''
-    def __init__(self, host, user, password, filename, log=None):
-        self.filename = filename
-
-        @retry(ftplib.error_temp, tries=4, delay=3, backoff=2, logger=log)
-        def ftp_connect_with_retry(host, user, password):
-            return ftplib.FTP_TLS(host, user, password)
-
-        self.connection = ftp_connect_with_retry(host, user, password)
-
-    def csv_reader(self):
-        # Just store the lines in memory
-        # It's non-ideal but neither classes support coroutine send/yield
-        lines = []
-        def callback(line):
-            lines.append(line)
-
-        self.connection.retrlines('RETR {}'.format(self.filename), callback)
-
-        r = csv.DictReader(lines)
-        return r
+    run(args)
```

Comment:
At this point, you've refactored everything down so well that ``main()`` doesn't actually do anything. That's good... it means we can get rid of it! I'd recommend either changing the import to ``from crmprtd.wmb_dir.download import run as main``, or directly calling ``run()`` instead of ``main()``. I'd prefer the latter.

And you can apply this suggestion to each of the other scripts as well.